crates/server/src/store/blocklist/authority.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2015-2022 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// https://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// https://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use std::{io, path::Path};
+
+use tracing::{debug, info, trace};
+
+use crate::{
+    authority::{
+        Authority, LookupError, LookupObject, LookupOptions, MessageRequest, UpdateResult, ZoneType,
+    },
+    proto::{
+        op::{Query, ResponseCode},
+        rr::{RData, LowerName, Name, Record, RecordType, rdata::A},
+    },
+    server::RequestInfo,
+    store::blocklist::BlockListConfig,
+};
+
+use crate::resolver::lookup::Lookup;
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::prelude::*;
+
+/// An authority that will resolve queries against one or more block lists.  The typical use case will be to use this in a chained
+/// configuration before a forwarding or recursive resolver:
+///
+///   [[zones]]
+///   zone = "."
+///   zone_type = "hint"
+///   stores = [{ type = "blocklist", lists = ["default/bl.txt", "default/bl2.txt"]}, { type = "recursor", roots = "default/root.zone"}]
+///
+/// Note: the order of the stores is important: the first one specified in the store list will be the first consulted.  Subsequent stores
+/// will only be consulted if each prior store returns Err(LookupError::NotHandled) in response to the query.
+
```

Comment:
minor, this extra newline might break the RustDoc from being associated, otherwise it's probably just a minor issue.

Reply:
Removed the extra line.